emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -67,20 +68,21 @@ def execute_and_time_function(func: t.Callable[[], bool]):
         logging.info(f"Function '{function_name}' executed successfully in {elapsed_ms:.2f} ms.")
 
         # Verification: Adjusted Query to Match Document Structure
-        timeseries_db = builtin_ts.get_timeseries_db(key="stats/dashboard_time")
-        
+        timeseries_db = esta.TimeSeries.get_time_series(None)
 
-        query = {
-            "metadata.key": "stats/dashboard_time",
-            "data.name": function_name,
+        # Define the time range and additional filters
+        time_query = {
             "data.ts": {"$gte": timestamp, "$lte": timestamp},
-            "data.reading": {"$gte": elapsed_ms, "$lte": elapsed_ms} 
+            "data.name": function_name,
+            "data.reading": {"$gte": elapsed_ms, "$lte": elapsed_ms}
         }
-        
-        # Retrieve the most recent document for the function
-        stored_document = timeseries_db.find_one(
-            query,
-            sort=[("data.ts", -1)]
+
+        # Retrieve the first matching entry using the get_first_entry method
+        stored_document = timeseries_db.get_first_entry(
```

Comment:
I don't think we should use `get_first_entry` here - just because you wrote `find_one` in the old code doesn't mean that you should change `get_first_entry` in the new code.

This is not actually even correct because if you test multiple functions, you will have multiple entries in the database. So the first entry is not the one you should be checking every time.

Reply:
@shankari Would using `get_entry_at_ts` be a better alternative? Since the functions run at diff ts (unless its parallel)

Reply:
Should we implement find_one on the aggregate timeseries?
What's the advantage of doing that vs using `get_first_entry` ?